### PR TITLE
Clean up disk space for coverage job

### PR DESCRIPTION
**Pull request by VenelinMartinov**
We saw some failures in AWS: https://github.com/pulumi/pulumi-aws/actions/runs/8076447038/job/22065470841

tested in https://github.com/pulumi/pulumi-aws/actions/runs/8079968659/job/22075523684?pr=3507

### CODE REVIEW

**Review by iwahbe — APPROVED** [review]
One request, but otherwise this looks good.

File: provider-ci/test-workflows/cloudflare/.github/workflows/master.yml [hunk 1]
```diff
@@ -138,6 +138,11 @@ jobs:
     needs: prerequisites
     runs-on: ubuntu-latest
     steps:
+    - name: Free Disk Space (Ubuntu)
+      uses: jlumbroso/free-disk-space@main
```
[the remark this answers is not in the document]
Comment (iwahbe) [reply provider-ci/test-workflows/cloudflare/.github/workflows/master.yml:142]: Automerged 😞

[the remark this answers is not in the document]
Comment (VenelinMartinov) [reply provider-ci/test-workflows/cloudflare/.github/workflows/master.yml:142]: mb, copied this from the repo, will pin it separately everywhere.

[the remark this answers is not in the document]
Comment (iwahbe) [reply provider-ci/test-workflows/cloudflare/.github/workflows/master.yml:142]: It's all good. Thanks for quickly following up!

**Pull request merged**